packages/features/calendars/weeklyview/components/event/EventList.tsx

```diff
@@ -20,110 +21,64 @@ export function EventList({ day }: Props) {
+    });
+  }, [events, day]);
 
-          const eventDuration = eventEnd.diff(eventStart, "minutes");
+  const layoutMap = useMemo(() => {
+    const layouts = calculateEventLayouts(dayEvents);
+    return createLayoutMap(layouts);
+  }, [dayEvents]);
 
-          const eventStartHour = eventStart.hour();
-          const eventStartDiff = (eventStartHour - (startHour || 0)) * 60 + eventStart.minute();
-          const nextEvent = eventsArray[idx + 1];
-          const prevEvent = eventsArray[idx - 1];
+  // Find which overlap group the hovered event belongs to
+  const hoveredEventLayout = hoveredEventId ? layoutMap.get(hoveredEventId) : null;
+  const hoveredGroupIndex = hoveredEventLayout?.groupIndex ?? null;
 
-          if (!longestRef.current) {
-            longestRef.current = {
-              idx,
-              start: eventStart.toDate(),
-              end: eventEnd.toDate(),
-              duration: eventDuration,
-            };
-          } else if (
-            eventDuration > longestRef.current.duration &&
-            eventStart.isBetween(longestRef.current.start, longestRef.current.end)
-          ) {
-            longestRef.current = {
-              idx,
-              start: eventStart.toDate(),
-              end: eventEnd.toDate(),
-              duration: eventDuration,
-            };
-          }
-          // By default longest event doesnt have any styles applied
-          if (longestRef.current.idx !== idx) {
-            if (nextEvent) {
-              // If we have a next event
-              const nextStart = dayjs(nextEvent.start);
-              // If the next event is inbetween the longest start and end make 65% width
-              if (nextStart.isBetween(longestRef.current.start, longestRef.current.end)) {
-                zIndex = 65;
-                marginLeft = "auto";
-                right = 4;
-                width = width / 2;
-
-                // If not - we check to see if the next starts within 5 mins of this event - allowing us to do side by side events whenwe have
-                // close start times
-              } else if (nextStart.isBetween(eventStart.add(-5, "minutes"), eventStart.add(5, "minutes"))) {
-                zIndex = 65;
-                marginLeft = "auto";
-                right = 4;
-                width = width / 2;
-              }
-            } else if (prevEvent) {
-              const prevStart = dayjs(prevEvent.start);
+  return (
+    <>
+      {dayEvents.map((event) => {
+        const layout = layoutMap.get(event.id);
+        if (!layout) return null;
 
-              // If the next event is inbetween the longest start and end make 65% width
+        const eventStart = dayjs(event.start);
+        const eventEnd = dayjs(event.end);
+        const eventDuration = eventEnd.diff(eventStart, "minutes");
+        const eventStartHour = eventStart.hour();
+        const eventStartDiff = (eventStartHour - (startHour || 0)) * 60 + eventStart.minute();
```

Comment:
instead of initializing dayjs on every render. we should pre compute this calculation above